build-system/eslint-rules/no-unload-listener.js

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+module.exports = function(context) {
+  const listenCall = 'CallExpression[callee.name=listen]';
```

Comment:
Add `listenOnce` and `listenOncePromise` as well. For example, here is a use:

https://github.com/ampproject/amphtml/blob/dc8c09499606b5d935dbe8fa82a57b59b24eb544/extensions/amp-viewer-integration/0.1/amp-viewer-integration.js#L225

It seems like `beforeUnload` also blocks the bfcache, so would want to add that here as well. I don't think this usage would be caught by lint, unless it runs after inlining of constants:

https://github.com/ampproject/amphtml/blob/dc8c09499606b5d935dbe8fa82a57b59b24eb544/src/service/performance-impl.js#L39